test/data/multiline.csv

```diff
@@ -0,0 +1,32 @@
+// this file should be loaded with quoting enabled
+// and it should load without problems by the TextLoader
+id,description,animal
```

Comment:
So here are several examples that the TextLoader should be able to load according to the RFC and to how it has worked in the past. Please, let me know if I should add more cases.

Also, from this [thread](https://github.com/dotnet/machinelearning/pull/4584#discussion_r360245121), I think it's better not to support these:

**Escaping `\"` inside quoted fields.** There are 3 main reasons to not support this:
1. The RFC 4180 doesn't support it. Quotes should be escaped using ""
2. Excel doesn't really support it consistently. If I have a CSV with `19,"this has an \"scaped quote\" right"` the first one will not be escaped but the second one does.
3. Among the datasets provided by ModelBuilder team some of them have the characters `\"` but they're not meant to escape the `"`. E.g., there are quoted fields that end with those characters `"blabla\"` and so the last `"` is supposed to be the closing of the field and the `\` is supposed to be part of the field. So escaping this would cause errors for those datasets. To implement this would require to add yet another option to TextLoader, in order to avoid those errors.

**Accept `True, "asdf,123` even when quoting is enabled** There are 2 main reasons for not doing this:
1. The RFC doesn't accept this.
2. In ML.NET we filter out the spaces at the beginning of the field, so there's no difference between that and `True,"asdf,123` which would be an error. So code-wise it would be weird to accept this particular case.

In general, I wouldn't accept these cases unless there were a lot of people asking for them.

Reply:
Would adding support for alternative escaping methods in the future be difficult? Slash escaping is [common enough](https://github.com/pandas-dev/pandas/blob/7456fc52e1546d263b88ba5772c0324f91ff0804/doc/source/user_guide/io.rst#quoting-and-escape-characters). Pandas (generally used by scikit-learn) exposes an `escapechar` parameter to allow for this.

Reply:
As discussed offline, I'll do this in an upcoming PR.

---
In reply to: [425735487](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425735487) [](ancestors = 425735487)